scripts/validation/validate-versions-staged-files.js

```diff
@@ -0,0 +1,76 @@
+const { readFile } = require("fs").promises;
+const { promisify } = require("util");
+const { join, dirname } = require("path");
+const { exec } = require("child_process");
+const { parseStringPromise } = require("xml2js");
+
+const execAsync = promisify(exec);
+
+preCommit().catch(error => {
+    console.error(error);
+    process.exit(1);
+});
+
+async function preCommit() {
+    const [{ stdout: lernaPackages }, { stdout: stagedFiles }] = await Promise.all([
+        execAsync("npx lerna ls --json --all"),
+        execAsync("git diff --staged --name-only")
+    ]);
+    const packages = JSON.parse(lernaPackages.trim());
+    const staged = stagedFiles.trim().split("\n");
+    const changedWidgetPackages = packages
+        .filter(({ location }) =>
+            staged.some(
+                changedFilePath =>
+                    dirname(join(process.cwd(), changedFilePath)).includes(location) &&
+                    changedFilePath
+                        .split("/")
+                        .pop()
+                        .match(/package\.(json|xml)$/)
+            )
+        )
+        .filter(({ location }) => location.match(/(pluggable|custom)Widgets/));
```

Comment:
Nit: You could do this `filter` first since it's less expensive than the previous one.

Reply:
Changed 👍